sdk/core/Azure.Core/src/Shared/AzureSasCredentialSynchronousPolicy.cs

```diff
@@ -29,13 +31,33 @@ public override void OnSendingRequest(HttpMessage message)
             {
                 signature = signature.Substring(1);
             }
+
             if (!query.Contains(signature))
             {
-                query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
+                // Get the signature history so we know which string to replace in the query if it was previously set.
+                if (message.TryGetProperty(typeof(AzureSasSignatureHistory), out object? previousSignature) &&
+                    previousSignature is string previousSignatureString &&
+                    query.Contains(previousSignatureString))
+                {
+                    query = query.Replace(previousSignatureString, signature);
+                }
+                else
+                {
+                    // This is the first time we're setting the signature, so we need to add it to the query.
+                    query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
```

Comment:
nit: string interpolation

Reply:
With simple concatenations, I think interpolation is actually slightly slower. Or at least the difference is negligable.

Reply:
It shouldn't be slower - the compiler will optimize as needed.
https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6/